src/pyodide/internal/tar.js

```diff
@@ -0,0 +1,83 @@
+import { default as Reader } from "pyodide-internal:packages_tar_reader";
+
+const decoder = new TextDecoder();
+function decodeString(buf) {
+  const nullIdx = buf.indexOf(0);
+  if (nullIdx >= 0) {
+    buf = buf.subarray(0, nullIdx);
+  }
+  return decoder.decode(buf);
+}
+function decodeField(buf, offset, size) {
+  return decodeString(buf.subarray(offset, offset + size));
+}
+function decodeNumber(buf, offset, size) {
+  return parseInt(decodeField(buf, offset, size), 8);
+}
+
+function decodeHeader(buf) {
+  const nameBase = decodeField(buf, 0, 100);
+  const namePrefix = decodeField(buf, 345, 155);
+  const path = namePrefix + nameBase;
+  const mode = decodeNumber(buf, 100, 8);
+  const size = decodeNumber(buf, 124, 12);
+  const modtime = decodeNumber(buf, 136, 12);
+  const type = Number(String.fromCharCode(buf[156]));
+  return {
+    path,
+    name: path,
+    mode,
+    size,
+    modtime,
+    type,
+    parts: [],
+    children: undefined,
+  };
+}
+
+export function tarInfo() {
```

Comment:
Nitpicking so feel free to ignore.
```suggestion
export function parseTarInfo() {
```